rpc-csharp/src/Runtime/MessageDispatcher.cs

```diff
@@ -7,15 +7,34 @@
 
 namespace rpc_csharp
 {
-    public class AckHelper
+    public struct ParsedMessage
     {
+        public RpcMessageTypes messageType;
+        public object message;
+        public uint messageNumber;
+
+        public ParsedMessage(RpcMessageTypes messageType, object message, uint messageNumber)
+        {
+            this.messageType = messageType;
+            this.message = message;
+            this.messageNumber = messageNumber;
+        }
+    }
```

Comment:
```suggestion
    public readonly struct ParsedMessage
    {
        public readonly RpcMessageTypes messageType;
        public readonly object message;
        public readonly uint messageNumber;

        public ParsedMessage(RpcMessageTypes messageType, object message, uint messageNumber)
        {
            this.messageType = messageType;
            this.message = message;
            this.messageNumber = messageNumber;
        }
    }
```

Reply:
I think that that `struct` could be defined as `readonly` so we can be sure that C# is not creating any defensive copy